src/core/agent/services/connectionService.ts

```diff
@@ -359,44 +360,54 @@ class ConnectionService extends AgentService {
         await this.createConnectionMetadata(contact.id, {
           alias: contact.alias,
           oobi: contact.oobi,
+          groupId: contact.groupCreationId,
+          createdAtUTC: contact.createdAt,
         });
       }
     }
   }
 
   @OnlineOnly
-  async resolveOobi(url: string, waitForCompletion = true): Promise<any> {
-    const startTime = Date.now();
+  async resolveOobi(
+    url: string,
+    waitForCompletion = true
+  ): Promise<{
+    op: Operation & { response: State };
+    connection?: Contact;
+    alias: string;
+  }> {
     const alias = new URL(url).searchParams.get("name") ?? randomSalt();
-    let operation;
+    let connection;
+    let operation: Operation & { response: State };
     if (waitForCompletion) {
-      operation = await waitAndGetDoneOp(
+      operation = (await waitAndGetDoneOp(
         this.props.signifyClient,
         await this.props.signifyClient.oobis().resolve(url, alias),
         5000
-      );
+      )) as Operation & { response: State };
+      if (!operation.done) {
+        throw new Error(ConnectionService.FAILED_TO_RESOLVE_OOBI);
+      }
     } else {
-      operation = await waitAndGetDoneOp(
-        this.props.signifyClient,
-        await this.props.signifyClient.oobis().resolve(url, alias),
-        2000 - (Date.now() - startTime)
-      );
-    }
-    if (!operation.done && !waitForCompletion) {
+      operation = await this.props.signifyClient.oobis().resolve(url, alias);
       const pendingOperation = await this.operationPendingStorage.save({
         id: operation.name,
         recordType: OperationPendingRecordType.Oobi,
       });
-
       this.props.eventEmitter.emit<OperationAddedEvent>({
         type: EventTypes.OperationAdded,
         payload: { operation: pendingOperation },
       });
-    } else if (!operation.done) {
-      throw new Error(ConnectionService.FAILED_TO_RESOLVE_OOBI);
     }
-    const oobi = { ...operation, alias };
-    return oobi;
+
+    if (operation.response && operation.response.i) {
```

Comment:
I think we can move this to the `waitForCompletion` block after the `if (!operation.done)` block.

Also, why aren't we storing `groupCreationId` here?

Reply:
if so, it won't create a contact for a normal connection because that case we pass `multiSigInvite=false`

Reply:
But we are creating the contact for a normal connection once the long running operation completes for the OOBI!

This call will throw an error in Signify right now if the long running operation isn't finished running.

Reply:
I see you made the change